libs/framework/src/manifest.c

```diff
@@ -24,271 +24,291 @@
-				status = CELIX_FILE_IO_EXCEPTION;
-				framework_logIfError(celix_frameworkLogger_globalLogger(), status, NULL, "Manifest '%s' line too long", filename);
-				break;
-			}
-			if (len > 0 && lbuf[len - 1] == '\r') {
-				--len;
-			}
-			if (len == 0 && skipEmptyLines) {
-				continue;
-			}
-			skipEmptyLines = false;
-
-			if (strlen(name) == 0) {
-				
-				if ((tolower(lbuf[0]) == 'n') && (tolower(lbuf[1]) == 'a') &&
-					(tolower(lbuf[2]) == 'm') && (tolower(lbuf[3]) == 'e') &&
-					(lbuf[4] == ':') && (lbuf[5] == ' ')) {
-					name[0] = '\0';
-					strncpy(name, lbuf+6, len - 6);
-					name[len - 6] = '\0';
-				} else {
-					status = CELIX_FILE_IO_EXCEPTION;
-					framework_logIfError(celix_frameworkLogger_globalLogger(), status, NULL, "Manifest '%s' invalid format", filename);
-					break;
-				}
-
-				if (fpeek(file) == ' ') {
-					int newlen = len - 6;
-					lastline[0] = '\0';
-					strncpy(lastline, lbuf+6, len - 6);
-					lastline[newlen] = '\0';
-					continue;
-				}
-			} else {
-				int newlen = strlen(lastline) + len;
-				char buf[512];
-				buf[0] = '\0';
-				strcpy(buf, lastline);
-				strncat(buf, lbuf+1, len - 1);
-				buf[newlen] = '\0';
+    celix_autoptr(FILE) file = fopen(filename, "r");
+    if (file != NULL) {
+        status = manifest_readFromStream(manifest, file);
+    } else {
+        status = CELIX_ERROR_MAKE(CELIX_FACILITY_CERRNO,errno);
+    }
 
-				if (fpeek(file) == ' ') {
-//					lastline = realloc(lastline, strlen(buf) + 1);
-					lastline[0] = '\0';
-					strcpy(lastline, buf);
-					continue;
-				}
-				name[0] = '\0';
-				strcpy(name, buf);
-				name[strlen(buf)] = '\0';
-			}
+    if (status != CELIX_SUCCESS) {
+        celix_err_pushf("Cannot read manifest %s: %s", filename, celix_strerror(status));
+    }
 
-			attributes = hashMap_get(manifest->attributes, name);
-			if (attributes == NULL) {
-				attributes = properties_create();
-				hashMap_put(manifest->attributes, strdup(name), attributes);
-			}
-			manifest_readAttributes(manifest, attributes, file);
+    return status;
+}
 
-			name[0] = '\0';
-			skipEmptyLines = true;
-		}
-		fclose(file);
-	} else {
-		status = CELIX_FILE_IO_EXCEPTION;
-	}
+celix_status_t manifest_readFromStream(manifest_pt manifest, FILE* stream) {
+    char lbuf[512];
+    char *bytes = lbuf;
 
-    if (status != CELIX_SUCCESS) {
-        fw_logCode(celix_frameworkLogger_globalLogger(), CELIX_LOG_LEVEL_ERROR, status, "Cannot read manifest");
+    // get file size
+    if(fseek(stream, 0L, SEEK_END) == -1) {
+        return CELIX_ERROR_MAKE(CELIX_FACILITY_CERRNO,errno);
+    }
+    long int size = ftell(stream);
+    if (size < 0) {
+        return CELIX_ERROR_MAKE(CELIX_FACILITY_CERRNO,errno);
+    } else if (size >= INT_MAX) {
+        celix_err_pushf("Manifest error: file too large - %ld", size);
+        return CELIX_BUNDLE_EXCEPTION;
+    }
+    rewind(stream);
+
+    celix_autofree char* largeBuf = NULL;
+    if (size+1 > sizeof(lbuf)) {
```

Comment:
nitpick: you have `lbuf` and `largeBuf`, I assume that `lbuf` means local buf. But in this case for clearity I would prefer something like `stackBuf` and `headBuf`